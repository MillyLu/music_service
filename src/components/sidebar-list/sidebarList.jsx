import { useState, useEffect} from 'react';
import * as Styled from './styles';
import { SidebarItem } from "../sidebar-item/sidebarItem";
import { SidebarSkeleton } from '../sidebar-skeleton/sidebarSkeleton';



export function SidebarList() {
    const [skeleton, setSkeleton] = useState(false);
  
    useEffect(() => {
        setSkeleton(true);
        setTimeout(async () => {

            setSkeleton(false);

        }, 300);
    }, []);
 
    return(
        <Styled.SidebarList>
            {!skeleton ?
            (<>
                <SidebarItem image = "playlist01.png" href="/playlist/1" />
                <SidebarItem image = "playlist02.png" href="/hits/1" />
                <SidebarItem image = "playlist03.png" href="/indi/1" />
            </>)
                
                :
            (<>
                <SidebarSkeleton />
                <SidebarSkeleton />
                <SidebarSkeleton />
            </>)
                

            }
        </Styled.SidebarList>
    )
}