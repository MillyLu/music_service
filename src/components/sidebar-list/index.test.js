/* eslint-disable no-undef */
import * as ShallowRenderer from 'react-shallow-renderer';
// import { ShallowRenderer, createRenderer } from 'react-shallow-renderer';
import { SidebarList } from './sidebarList';



describe('<SidebarList/>', () => {
        it('should render successfully and match snapshot', () => {
            const shallowRenderer = ShallowRenderer.createRenderer();
            const renderer = shallowRenderer;
            renderer.render(<SidebarList />);
            const result = renderer.getRenderOutput();
            expect(result).toMatchSnapshot();
          })
      });