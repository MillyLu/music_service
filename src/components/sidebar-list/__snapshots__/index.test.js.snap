// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SidebarList/> should render successfully and match snapshot 1`] = `
<styled.div>
  <React.Fragment>
    <SidebarItem
      href="/playlist/1"
      image="playlist01.png"
    />
    <SidebarItem
      href="/hits/1"
      image="playlist02.png"
    />
    <SidebarItem
      href="/indi/1"
      image="playlist03.png"
    />
  </React.Fragment>
</styled.div>
`;
